File: Lab/Lab 1/lab1.java
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class lab1 {
    public static void main(String[] args) {
        int[] nums = {0,1,8,7,3,9};
        int target = 16;
        int[] solution = new int[2];

        Map <Integer, Integer> numbers = new HashMap<Integer, Integer>();

        for (int i = 0; i < nums.length; i++){
            int total = 0;

            total = target - nums[i];

            if (!numbers.containsKey(total)){
                numbers.put(nums[i], i);
            }
            else {
                solution[0] = numbers.get(total);
                solution[1] = i;
            }
        }
        System.out.println(Arrays.toString(solution) + "\nIndices: " + solution[0] + " and " + solution[1]);
    }
}

File: Lab/Lab 1/lab1.py
nums = [0,1,8,7,3,9]
target = 16
solution = []

dictionary = {
}

length = len(nums)

for i in range(length):
    total = 0
    total = target - nums[i]
    
    if dictionary.get(total) is None:
        dictionary.update({nums[i]:i})
    else:
        solution.append(dictionary.get(total))
        solution.append(i)

print(solution)
print("Indices " + str(solution[0]) + " and " + str(solution[1]))
